dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatClientChatCompletionService.cs

```diff
@@ -67,7 +67,7 @@ public async Task<IReadOnlyList<ChatMessageContent>> GetChatMessageContentsAsync
                 .Skip(currentSize)
                 .Select(m => ChatCompletionServiceExtensions.ToChatMessageContent(m)));
 
-        return completion.Choices.Select(m => ChatCompletionServiceExtensions.ToChatMessageContent(m, completion)).ToList();
+        return completion.Messages.Select(m => ChatCompletionServiceExtensions.ToChatMessageContent(m, completion)).ToList();
```

Comment:
This doesn't look right. The GetChatMessageContentsAsync method is returning a single message for each choice, but this ends up treating all of the response messages in a single choice each as being their own unique choice. I think it needs to be more like this:
```C#
for (int i = 0; i < completion.Messages.Count - 1; i++)
{
    chatHistory.Add(ChatCompletionServiceExtensions.ToChatMessageContent(completion.Messages[i], completion));
}

List<ChatMessageContent> content = [];
if (completion.Messages.Count != 0)
{
    content.Add(ChatCompletionServiceExtensions.ToChatMessageContent(completion.Messages[0], completion));
}
return content;
```
in order to bridge the differences in the models.